code/go/0chain.net/smartcontract/minersc/handler.go

```diff
@@ -126,6 +117,40 @@ func (msc *MinerSmartContract) GetMinerListHandler(ctx context.Context, params u
 	}, nil
 }
 
+// GetMinerGeolocationsHandler returns a list of all miners latitude and longitude
+func (msc *MinerSmartContract) GetMinerGeolocationsHandler(ctx context.Context, params url.Values, balances cstate.StateContextI) (interface{}, error) {
+	var (
+		offsetString = params.Get("offset")
+		limitString  = params.Get("limit")
+		activeString = params.Get("active")
+	)
+
+	offset, limit, err := getOffsetLimitParam(offsetString, limitString)
+	if err != nil {
+		return nil, err
+	}
+
+	filter := event.MinerQuery{}
+	if activeString != "" {
+		active, err := strconv.ParseBool(activeString)
+		if err != nil {
+			return nil, common.NewErrBadRequest("active parameter is not valid")
+		}
+		filter.Active = null.BoolFrom(active)
+	}
+
+	if balances.GetEventDB() == nil {
```

Comment:
Move this nil eventDB checking upper to avoid unnecessary works.

Reply:
yes, got it.

Reply:
done